server/src/main/java/org/elasticsearch/action/fieldcaps/IndexFieldCapabilities.java

```diff
@@ -50,7 +50,7 @@
 
     IndexFieldCapabilities(StreamInput in) throws IOException {
         this.name = in.readString();
-        this.type = in.readString();
+        this.type = in.readString().intern();
```

Comment:
I believe `String.intern` is usually not recommended because you have no control over the size of the string pool. It used to be pretty bad before Java 8 where the string pool was finally moved to the heap from PermGen. I think the better approach is too use a ConcurrentMap with Weak/SoftReferences or an actual `Cache` if it's available.

Reply:
https://github.com/FasterXML/jackson-databind/blob/2.13/src/main/java/com/fasterxml/jackson/databind/util/LRUMap.java#L24
For example, the approach which is used in Jackson for caching

Reply:
@arteam True true, in this one instance I figured I could get away with it because we know for a fact that these strings will be in the pool (these are field types and the field mapper objects all have them as concrete strings in the source). 
I could certainly see the point of using a `static` constant CHM and putting the interned strings in therefor performance which will probably be quite a bit faster even in recent JDK. It's just extra code and maybe a bit of a risk in case the type ever stops coming from a constant pool of strings.

Reply:
@arteam I made this a little more "high-tech" now by putting a CHM in front of intern :) Should be way faster now with the same result as before. Let me know what you think :)